penguin/web/services/provider_catalog.py

```
@@ -312,6 +315,95 @@ def codex_oauth_cached_provider_models(
     return {"openai": models} if models else {}
 
 
+def apply_cached_codex_model_metadata(model_config: Any) -> None:
```

Ruff (0.15.20)

[warning] Too many return statements (9 > 6)

(PLR0911)